### DISCUSSION TITLE
es/guides/sahi-tiled-inference/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14686] [Docs]
# es/guides/sahi-tiled-inference/

Más información sobre cómo implementar YOLOv8 con SAHI para la inferencia segmentada. Optimice el uso de la memoria y mejore la precisión de la detección para aplicaciones a gran escala.

https://docs.ultralytics.com/es/guides/sahi-tiled-inference/

**Comment by wisrovi** [comment]
Hola, 

estoy necesitando procesar por lotes, pero no tengo las imagenes almacenadas, sino que vienen de un video

```python
cap = cv2.VideoCapture(video)

success, im0 = cap.read()
success, im1 = cap.read()
success, im2 = cap.read()

detection_model = AutoDetectionModel.from_pretrained(
        model_type="rtdetr",  
        model_path=MODEL_DETECTOR,
        confidence_threshold=0.15,
        device="cuda:0",  # or 'cuda:0'  or "cpu"
    )


# esto si funciona
result = get_prediction(im0, detection_model)

# esto NO funciona, pero necesito poder procesar por lotes en lugar de una en una, pensè en:
images_chain_thread = [im0, im1, im2] 
result = get_prediction(images_chain_thread, detection_model)
```

Es posible procesar por lotes de esta manera?

me darian alguna orientacion por favor

**Reply by glenn-jocher** [reply]
@wisrovi hola,

Para procesar imágenes por lotes desde un video, necesitarás adaptar tu enfoque ya que `get_prediction` no admite directamente una lista de imágenes. Una solución es procesar cada cuadro individualmente y luego combinar los resultados. Alternativamente, puedes explorar la integración con SAHI para manejar inferencia segmentada, lo cual es útil para imágenes grandes o de alta resolución. Para más detalles sobre SAHI y su integración con YOLOv8, visita [SAHI Tiled Inference Guide](https://docs.ultralytics.com/guides/sahi-tiled-inference/).
